Test/Resolve-WhiskeyTaskPathParameter.Tests.ps1

```diff
@@ -358,8 +408,188 @@ Describe ('Resolve-WhiskeyTaskPathParameter.when a path uses different case to t
             ThenPipelineSucceeded
         }
     }
 }
+
+Describe 'Resolve-WhiskeyTaskPath.when working directory isn''t build root and using a non-existent relative path inside the build root' {
+    It 'should allow the path' {
+        Init
+        GivenDirectory 'subdir'
+        WhenRunningTask 'ValidateOptionalNonExistentPathTask' -Parameter @{ 'Path' = '..\NewFile.txt'; 'WorkingDirectory' = 'subdir' }
+        ThenTaskCalled -WithParameter @{ 'Path' = (Resolve-RelativePath '..\Newfile.txt') }
+    }
+}
+
+Describe 'Resolve-WhiskeyTaskPathParamter.when file should get created' {
+    It 'should create the item' {
+        Init
+        # We use a directory since `New-Item` defaults to creating files.
+        WhenRunningTask 'CreateMissingFileTask' -Parameter @{ 'Path' = 'dir\One.txt','Two.txt' }
+        ThenTaskCalled -WithParameter @{ 'Path' = (Resolve-RelativePath 'dir\One.txt','Two.txt') }
+        Join-Path -Path $testRoot -ChildPath 'dir\One.txt' | Should -Exist
+        Join-Path -Path $testRoot -ChildPath 'Two.txt' | Should -Exist
+    }
+}
+
+Describe 'Resolve-WhiskeyTaskPathParamter.when directory should get created' {
+    It 'should create the item' {
+        Init
+        # We use a directory since `New-Item` defaults to creating files.
+        WhenRunningTask 'CreateMissingDirectoryTask' -Parameter @{ 'Path' = 'One','Two' }
+        ThenTaskCalled -WithParameter @{ 'Path' = (Resolve-RelativePath 'One','Two') }
+        Join-Path -Path $testRoot -ChildPath 'One' | Should -Exist
+        Join-Path -Path $testRoot -ChildPath 'Two' | Should -Exist
+    }
+}
+
+Describe 'Resolve-WhiskeyTaskPath.when item should get created but path type not given' {
+    It 'should fail' {
+        Init
+        WhenRunningTask 'CreateMissingItemwithPathTypeMissingTask' -Parameter @{ 'Path' = 'One.txt' } -ErrorAction SilentlyContinue
+        ThenTaskNotCalled
+        ThenThrewException -Pattern 'add a PathType property'
+    }
+}
+
+Describe 'Resolve-WhiskeyTaskPath.when path is current directory' {
+    It 'should pass' {
+        Init
+        WhenRunningTask 'ValidateMandatoryDirectoryTask' -Parameter @{ 'Path' = '.' }
+        ThenTaskCalled -WithParameter @{ 'Path' = ('.{0}' -f [IO.Path]::DirectorySeparatorChar) }
+    }
+}
+
+Describe 'Resolve-WhiskeyTaskPath.when path does not exist and some paths use wildcards and others do not' {
+    It 'should normalize directory separators' {
+        Init
+        WhenRunningTask 'ValidateMandatoryNonexistentFilesTask' -Parameter @{ 'Path' = @( 'file.txt','*.json','anotherfile.txt') }
+        $expectedPaths = & {
+            Resolve-RelativePath 'file.txt'
+            Resolve-RelativePath 'anotherfile.txt'
+        }
+        ThenTaskCalled -WithParameter @{ 'Path' = $expectedPaths }
+    }
+}
+
+Describe 'Resolve-WhiskeyTaskPath.when path does not exist and user is using directory separator from another OS' {
+    It 'should normalize directory separators' {
+        if( $IsWindows )

```

Comment:
Windows FS natively works with "/" as a separator. It's just less commonly used in paths (and some broken programs might not understand it's a valid path separator).

Reply:
Yes. The file system does, but I'm working under the assumption that people (like me) like to see the default separator for their OS. I've changed this to be a capability check instead of a platform check.